### PR TITLE
Update Caddy to v2.6.3

### LINKED ISSUE #14116: Login window is not closed
'Seen that on a preview environment, but it just confirmed to happen on prod as well.

The `/complete-auth` redirect seems to be broken. When opened with a `?message=success` query, it should post a message to the opener, which in turn will close the child window. Instead we see a 308 "redirect" to the same path, but without any query. 

Screen Shot 2022-10-21 at 14 28 06

To be investigated.

### PR DESCRIPTION
**Pull request by aledbf**
## Description

https://github.com/caddyserver/caddy/releases/tag/v2.6.3

fixes #14116
fixes #14119

## Release Notes
```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by vulkoingim — APPROVED** [review]

**Review by gtsiolis — APPROVED** [review]

**Review by mustard-mh — APPROVED** [review]
Tested in preview env with Github login, iframe auto closed.

**Review by Furisto — APPROVED** [review]

File: components/dashboard/conf/Caddyfile [hunk 1]
```diff
@@ -43,6 +43,10 @@
 
 	rewrite /schemas/gitpod-schema.json /static/schemas/gitpod-schema.json
 
+	# avoid multiple redirects to return a static html file to not loose
+	# window.opener and brake the automatic closing of the auth redirect
+	rewrite /complete-auth /complete-auth/index.html
```
Comment (AlexTugarev) [inline components/dashboard/conf/Caddyfile:48]: 🎉  nice!!!

**Comment by AlexTugarev** [comment]
Retesting with GH/GL/BB on a preview env...

**Review by AlexTugarev — APPROVED** [review]
Awesome!

**Pull request merged**